### PR TITLE
Get draft end point

### PR DESCRIPTION
**Pull request by tej99**

### CODE REVIEW

File: app/resources/messages.py [hunk 1]
```diff
@@ -125,7 +125,7 @@ def post(self):
         is_draft = False
         draft_id = None
         if 'msg_id' in post_data:
-            is_draft = DraftById().check_valid_draft(post_data['msg_id'])
+            is_draft = DraftModifyById().check_valid_draft(post_data['msg_id'])
```
Comment (gemmai95) [inline app/resources/messages.py]: Is this the best place to be checking whether something is a draft as it is common functionality for a number of endpoints?

File: tests/app/test_retriever.py [hunk 2]
```diff
@@ -28,28 +28,29 @@ def setUp(self):
             database.db.create_all()
             self.db = database.db
 
-    def populate_database(self, x=0, convo=False, draft=False, multiple_users=False):
+    def populate_database(self, x=0, single=True, convo=False, draft=False, multiple_users=False):
         with self.engine.connect() as con:
```
Comment (AndrewTorrance) [inline tests/app/test_retriever.py]: Variable called X ????????????  Please change it to something descriptive

File: tests/app/test_retriever.py [hunk 3]
```diff
@@ -28,28 +28,29 @@ def setUp(self):
...
+                    query = 'INSERT INTO status(label, msg_id, actor) VALUES("SENT", "{0}", "respondent.21345")'.format(
+                        msg_id)
+                    con.execute(query)
+                    query = 'INSERT INTO status(label, msg_id, actor) VALUES("INBOX", "{0}", "SurveyType")'.format(
+                        msg_id)
+                    con.execute(query)
+                    query = 'INSERT INTO status(label, msg_id, actor) VALUES("UNREAD", "{0}", "SurveyType")'.format(
+                        msg_id)
+                    con.execute(query)
                 if convo:
```
Comment (AndrewTorrance) [inline tests/app/test_retriever.py]: convo ? Can you make the variable name a bit more descriptive please

File: tests/app/test_retriever.py [hunk 4]
```diff
@@ -28,28 +28,29 @@ def setUp(self):
...
-                query = 'INSERT INTO status(label, msg_id, actor) VALUES("SENT", "{0}", "respondent.21345")'.format(
-                    msg_id)
-                con.execute(query)
-                query = 'INSERT INTO status(label, msg_id, actor) VALUES("INBOX", "{0}", "SurveyType")'.format(
-                    msg_id)
-                con.execute(query)
-                query = 'INSERT INTO status(label, msg_id, actor) VALUES("UNREAD", "{0}", "SurveyType")'.format(
-                    msg_id)
-                con.execute(query)
+                if single:
```
Comment (AndrewTorrance) [inline tests/app/test_retriever.py]: I am not sure why populate database is one big function . For discussion- Would it be easier to follow if we had seperate functions such as populate_database_single_user, populate_draft etc

**Pull request merged**